src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -66,30 +75,113 @@ public async Task<SecretResponseModel> GetAsync([FromRoute] Guid id)
+    [HttpGet("projects/{projectId}/{organizationId}/secrets")]
+    public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId, [FromRoute] Guid organizationId)
     {
-        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId);
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
+        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId, userId, accessClient);
+
         var responses = secrets.Select(s => new SecretResponseModel(s));
         return new SecretWithProjectsListResponseModel(secrets);
     }
 
-    [HttpPut("secrets/{id}")]
-    public async Task<SecretResponseModel> UpdateAsync([FromRoute] Guid id, [FromBody] SecretUpdateRequestModel updateRequest)
+    [HttpPost("organizations/{organizationId}/secrets")]
+    public async Task<SecretResponseModel> CreateSecretAsync([FromRoute] Guid organizationId, [FromBody] SecretCreateRequestModel createRequest)
     {
-        var result = await _updateSecretCommand.UpdateAsync(updateRequest.ToSecret(id));
+        var userId = _userService.GetProperUserId(User).Value;
+        var secret = createRequest.ToSecret(organizationId);
+        var result = await _createSecretCommand.CreateAsync(createRequest.ToSecret(organizationId), userId);
         return new SecretResponseModel(result);
     }
 
-    // TODO Once permissions are setup for Secrets Manager need to enforce them on delete.
-    [HttpPost("secrets/delete")]
-    public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids)
+    [HttpPut("secrets/{organizationId}/{id}")]
+    public async Task<SecretResponseModel> UpdateSecretAsync([FromRoute] Guid organizationId, [FromRoute] Guid id, [FromBody] SecretUpdateRequestModel updateRequest)
     {
-        var results = await _deleteSecretCommand.DeleteSecrets(ids);
+        var userId = _userService.GetProperUserId(User).Value;
+        var secret = updateRequest.ToSecret(id, organizationId);
+        var result = await _updateSecretCommand.UpdateAsync(secret, userId);
+        return new SecretResponseModel(result);
+    }
+
+    [HttpPost("secrets/{organizationId}/delete")]
+    public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([FromBody] List<Guid> ids, [FromRoute] Guid organizationId)
+    {
+        var userId = _userService.GetProperUserId(User).Value;
+        var results = await _deleteSecretCommand.DeleteSecrets(ids, userId, organizationId);
         var responses = results.Select(r => new BulkDeleteResponseModel(r.Item1.Id, r.Item2));
         return new ListResponseModel<BulkDeleteResponseModel>(responses);
     }
```

Comment:
Take a look at the `ProjectsController` and the `DeleteProjectCommand` we don't need to add the organizationID into the URL path and can simply obtain it when we fetch the secrets in the `DeleteSecretCommand`.